internal/pkg/ecs/ecs.go

```diff
@@ -232,6 +247,34 @@ func (c Client) StopDefaultClusterTasks(familyName string) error {
 	return c.ecsClient.StopTasks(taskIDs, ecs.WithStopTaskReason(taskStopReason))
 }
 
+// DecryptedSecrets returns the decrypted parameters from either SSM parameter store or Secrets Manager.
+func (c Client) DecryptedSecrets(secrets []*ecs.ContainerSecret) ([]EnvVar, error) {
+	var vars []EnvVar
+	for _, secret := range secrets {
+		parsed, err := arn.Parse(secret.ValueFrom)
+		if err != nil || parsed.Service == ssm.Namespace {
```

Comment:
The logic is confusing to me. We have two conditions here for `err` and `parsed.Service` 6 possible cases: 2 for `err` (nil or not nil) and 3 for `parsedService` (ssm.Namespace, secretsmanager.Namespace, or others).

I think we can combine some cases. For example, if `err != nil` that means we didn't parse successfully and we should probably error out in this situation. Otherwise we can do a switch for the `parsed.Service` and make API calls accordingly (for the default case which is other namespace value we should probably also error out).